### PR TITLE
Added property in Slider component to all for Slider Thumb removal

### PR DESCRIPTION
**Pull request by gitars**
Earlier, I wanted to see the addition of a progress bar, and recently it was brought up to me that the Slider component can easily be modified to be a progress bar. So basically, what I have here is the addition of a property to the Slider component that will remove the Thumb Slider, and you will simply be left with a progress bar. The way it works is, when the property, ThumbEnabled = true, nothing is changed. When it is false, the Thumb Slider is hidden and the ability to change the progress (from the UI) is disabled (you can still change the progress from the code/blocks). 

The reason I went with "ThumbEnabled" was because there is another property "ThumbPosition" so I figured they kind of go together.

Also, this is not a complete feature because it lacks Internationalization which I am still not 100% sure how to do. If anything else is missing, please let me know.

### CODE REVIEW

**Comment by marksherman** [comment]
I tested locally. Looks great to me! No runtime errors in companion or apk build.
Notes:
- tested thoroughly: the thumb handle is NOT touchable/draggable when invisible
- bringing the thumb handle back to visible instantly makes it interactive again, vice-versa
- The PositionChanged event still fires regardless of handle visibility (as expected)

I have an app that exercises the feature that I'll post to the google group.

File: appinventor/components/src/com/google/appinventor/components/common/YaVersion.java [hunk 1]
```diff
@@ -726,7 +726,9 @@ private YaVersion() {
 
   // For SLIDER_COMPONENT_VERSION 1:
   // - Initial version.
-  public static final int SLIDER_COMPONENT_VERSION = 1;
+  // For SLIDER_COMPONENT_VERSION 2:
+  // - Added the property to allow for the removal of the Thumb Slider
+  public static final int SLIDER_COMPONENT_VERSION = 2;
```
Comment (jisqyv) [inline appinventor/components/src/com/google/appinventor/components/common/YaVersion.java]: You also need to update YOUNG_ANDROID_VERSION. Search for it in the code, what you need to do is obvious (or should be :-) )

**Comment by jisqyv** [comment]
There is an interesting argument that we should implement a "real" progress bar, aka use the one provided by Android. By using the Android provided version the user will get a same "feel" of the version of Android running on her device. For example with Lollipop, it will be a material design version.

**Comment by jisqyv** [comment]
Perhaps #268 should be updated to provide both a spinner and the progress bar.

**Comment by gitars** [comment]
@jisqyv So, I think this will also provide the same look and feel as the Android Progress Bar, because the Slider component is an Android Seekbar, which is an extension of the Android Progress Bar. I think the argument could still be made for a Progress Bar component, but it would have all the same methods as the Slider component, which might be undesirable.

In regards to #268, it could use a progress bar, but then we would have to add a good number of methods to that progress bar functional, by again adding all the methods in the Slider component.

**Comment by jisqyv** [comment]
@gitars Good point.

File: appinventor/components/src/com/google/appinventor/components/common/YaVersion.java (hunk 1 above)
Comment (gitars) [inline appinventor/components/src/com/google/appinventor/components/common/YaVersion.java]: Good note, thanks! I updated it, hopefully that was all.

**Comment by halatmit** [comment]
Make sure the new property is internationalized.




== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty

**Comment by gitars** [comment]
Yup, I noted that above. I was hoping that @jisqyv could help me with the Internationalization of #268 and afterwards I would be more capable of doing so for this feature.

**Comment by halatmit** [comment]
See the comment in YaVersion at line 392 ff



== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty

**Comment by gitars** [comment]
@halatmit Yes, I have seen those docs before, but it seems a little undocumented to me. It says to add a message to OdeMessages, so I did that. Then it says, You then have to add a mapping to TranslationComponentProperties.java (or ..Methods or ..Events, etc). As I based what I added the ThumbPosition method, which is only in TranslationComponentProperty.java and TranslationDesignerProperties.java, I added a mapping for ThumbEnabled in those two files. 

Then, it says "If you edit the description of a component (but not yet a property,method or event of that component) you must also find and update the description in OdeMessages.java." But I did not edit the description of the component. 

That's all it says to do, and I don't know what else I am missing. If you have more information, that would be helpful.

**Comment by halatmit** [comment]
That sounds like you did the right thing. We'll look carefully when we do
the code review.  == Thanks




== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty

**Comment by jisqyv** [comment]
@gitars @halatmit I'll complete the review (and merge if no issues) tonight. So far it looks good.

**Comment by jisqyv** [comment]
LGTM. MERGED

@gitars I have squashed your changes to one commit, edited the commit message and MERGED the result into master. Note: I made two edits. I edited versioning.js to deal with the component update (a new requirement that isn't well documented) and I also updated the on-line documents. You can review the commit for details.

**Comment by gitars** [comment]
Awesome thanks!

**Pull request closed without merging**